test/extensions/filters/http/squash/squash_filter_test.cc

```diff
@@ -432,8 +432,9 @@ TEST_F(SquashFilterTest, TimerExpiresInline) {
   attachmentTimeout_timer_ = new NiceMock<Envoy::Event::MockTimer>(&filter_callbacks_.dispatcher_);
   EXPECT_CALL(*attachmentTimeout_timer_, enableTimer(config_->attachmentTimeout()))
       .WillOnce(Invoke([&](const std::chrono::milliseconds&) {
+        attachmentTimeout_timer_->enabled_ = true;
```

Comment:
Can this actually happen? I don't think so?

Reply:
I don't think the timer can be invoked literally under enabletimer.  My guess is that's the only hook the test author had for squash finishing under async client -> send, but I was mainly trying to preserve existing tests as much as possible.

Reply:
Do you mind just adding a TODO to fix this test to make it more realistic? Thank you.

/wait